### ISSUE TITLE
Icons dissapeared

### ISSUE THREAD

**Issue by AcidScorpion** [powerline/powerline#1674]
In latest version disappeared "clock" and "weather" icons
![screenshot 2016-10-30 14 28 51](https://cloud.githubusercontent.com/assets/23050688/19836606/2760a4b2-9eae-11e6-828b-932f8d71aea0.png)

**Comment by s-ol** [comment]
@AcidScorpion did you set a `top_theme` for that ext in `config.json`?

**Comment by ZyX-I** [comment]
New default configuration uses only symbols which are present in terminus pcf font or in powerlinesymbols font as these symbols are less likely to cause troubles like in #1652. Old theme is still there and it is still named `powerline`, but default is `powerline_terminus`. You may use `config.json/common/default_top_theme` or `config.json/ext/{ext}/top_theme` configuration to change this.

**Comment by ZyX-I** [comment]
(Also I honestly have no idea what “watch” symbol may be useful for (main content of time segment is pretty obvious) or how you can understand what most of those weather icons mean.)

**Comment by AcidScorpion** [comment]
Well, if I understand correctly, the main idea of this project is fancy console look of usual and necessary stuff. So, icons doing their job perfectly. And, obviously, should be. Otherwise, pure tmux, for example, doing its job well and I can configure, more-or-less with the same look.
 
About weather icons, hm... it's simply. Looks clear, I guess:
_powerline/config_files/themes/powerline.json_:
```json
"powerline.segments.common.wthr.weather": {
                        "args": {
                                "icons": {
                                        "day":           "〇",
                                        "blustery":      "⚑",
                                        "rainy":         "☔",
                                        "cloudy":        "☁",
                                        "snowy":         "❅",
                                        "stormy":        "☈",
                                        "foggy":         "≡",
                                        "sunny":         "☼",
                                        "night":         "☾",
                                        "windy":         "☴",
                                        "not_available": "�",
                                        "unknown":       "⚠"
                                }
                        }
                },
```
And I really disappointed, why in _powerline/config_files/themes/powerline_terminus.json_ they been replaces by the words. Latest terminus TTF font support all those characters from the box without patching
https://files.ax86.net/terminus-ttf
At least for Mac

P.S. Thanks for help!

**Comment by ZyX-I** [comment]
@AcidScorpion Terminus from upstream (http://terminus-font.sourceforge.net/) has only

- `foggy`: basically because used symbol has nothing to do with weather.
- `sunny`
- `not_available`: same as `foggy`, nothing to do with weather.

And that’s all. Checked by using ipython2, fontforge and archive from upstream: basically opened one font file (`font = fontforge.open('ter-u32n.bdf')`) and checked all glyphs by issuing `font[0xXXXX]` (where `U+XXXX` is unicode codepoint of the glyph in question). Most of time this resulted in “TypeError: no such glyph” (don’t ask me why this is a TypeError and not more idiomatic KeyError or, at least, IndexError). AFAIR I used something similar when first creating `unicode_terminus` theme (when creating `powerline_terminus` I simply copied `unicode_terminus` and did some changes).

Do not know what about other distributions, but main portage tree (Gentoo) contains only *this* terminus and not the ttf version.

**Comment by ZyX-I** [comment]
By the way, ttf upstream explicitly says

> If the application you want to use the font with supports the original Terminus Font, you should really use that one instead of this TTF version — it will most likely be a more pleasant experience for you.

**Comment by ZyX-I** [comment]
I must also ask how did you check that ttf version supports these glyphs: do remember that modern font rendering libraries can use glyphs from *another* font if the selected one has these glyphs missing. If you see some glyph on your screen this means basically nothing, to perform such checks first take an application which has font renderer without such capability or disable this feature via configuration (not sure when this is possible though). E.g. AFAIK rxvt-unicode terminal will not take glyphs from other fonts unless you explicitly list fonts to take missing glyphs from in its configuration.

**Comment by AcidScorpion** [comment]
I really don't know, how glyphs magic works, but suspect the main reason in terminal software. Personally, I'm using iTerm2 under OS X and GNOME Terminal under Fedora.

Under OS X TTF fonts is main (and may be one) way to install and use fonts. I've tried to get original terminus, patched them and convert it to TTF, but gave up because it's complicated. After I decided to find ready to use TTF fonts and found, that font from https://files.ax86.net/terminus-ttf/ looks good and show everything I need in powerline. Of course, TTF Terminus font maintainers made a remarks, because nobody know, how that font will work in any terminals.

Under Fedora 24 I've installed font and powerline from its repo, so this is not TTF version.
GNOME terminal:
![2016-10-31 11_44_29-fedora24 - vmware workstation 12 player non-commercial use only](https://cloud.githubusercontent.com/assets/23050688/19850281/5fa33666-9f60-11e6-8c11-d52e9eba267e.png)

rxvt-unicode:
![2016-10-31 11_45_10-fedora24 - vmware workstation 12 player non-commercial use only](https://cloud.githubusercontent.com/assets/23050688/19850282/61c4ca72-9f60-11e6-8268-bbef223a3190.png)

Screen from iTerm2 in first message

**Comment by ZyX-I** [comment]
Most of terminal emulators do not bother having their own font rendering library, using the existing ones. AFAIK on linux there are two main libraries: freetype (which is able to use fallback fonts and does so by default, configured via XML files) and libXfont (X11 font rasterisation library that does not depend on freetype unlike libXft; not the best choice, but it should always be there). All terminals I know use freetype (even rxvt-unicode; at least they link with it and not with libXfont), but apparently do this rather differently: konsole happens to render fallback glyphs too wide to fit into a screen cell, producing more or less obtrusive visual bugs; rxvt-unicode somehow tells freetype to ignore all configured fallbacks, except for those explicitly provided in *terminal emulator* configuration; not sure about other terminals. As you may see in your second screenshot, watch symbol is not in terminus even though GNOME terminal is able to display it.

Dunno why you are able to see symbol used to indicate “blustery” weather, guess somebody configured rxvt-unicode to use fallback fonts, but did not configure to also use PowerlineSymbols as a fallback (or you would see arrows). But the fact that GNOME terminal and rxvt-unicode have different shape of flags means that they have different fallback font. This just shows that idea “print characters to determine whether font contains them” is poor by design, better take some specialized font viewer or editor (like fontforge) and check there instead.

**Comment by AcidScorpion** [comment]
IMHO, if point only in _weather and clock icons_, so may be add additional parameter in
**powerline.segments.common.wthr.weather**
**powerline.segments.common.time.date**
and user will decide, what' better to use in his case - icons or text
rather than add/make new _top_theme_

**Comment by AcidScorpion** [comment]
Small update. To get visible all symbols from powerline_unicode7 top-level theme need to install _symbola_ font. For RH family, this is **gdouros-symbola-fonts-8.00-5.fc24.noarch**. Can be obtained from Fedora repo.
Helpful link : https://powerline.readthedocs.io/en/master/troubleshooting/linux.html

**Comment by ZyX-I** [comment]
@AcidScorpion `powerline_unicode7` top theme has a big problem: for it to work properly your terminal emulator, used program(s) (sometimes libraries they are using and not programs themselves, but e.g. Vim supports unicode directly and not via icu/etc) and libc need to support unicode-7.0 or higher standard. When theme was just created it was not usable, but situation tends to improve over time.

**Comment by AcidScorpion** [comment]
As I get working `powerline_unicode7` in RHEL 7, which has mostly not so fresh packages, seems it should work now almost everywhere.

**Comment by ZyX-I** [comment]
I have konsole-16.04.3 and it still has problems with those characters. Though it is not the latest version available in Gentoo (latest is 16.08.3), it is latest stable version. When I was writing `powerline_unicode7` AFAIR Vim did not support unicode-7 (though do not remember exactly, but unicode-7 was released in June 2014 and theme was added in December 2014, when did Vim add support is not so easy to determine: unlike Neovim which now simply contains a copy of unicode files in repository Vim has this in `mbyte.c` alongside with other code modified for other reasons), though I am sure now it does support this standard.
